app/uk/gov/hmrc/tariffclassificationfrontend/service/CommodityCodeService.scala

```diff
@@ -16,26 +16,56 @@
 
 package uk.gov.hmrc.tariffclassificationfrontend.service
 
-import javax.inject.Singleton
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.models.CommodityCode
 
 import scala.io.Source
 
 @Singleton
-class CommodityCodeService() {
+class CommodityCodeService @Inject()(appConfig: AppConfig) {
 
+  private lazy val dateTimeFormatter = DateTimeFormatter.ofPattern("yyy-MM-dd HH:mm:ss")
   private lazy val padLimit = 10
 
-  def checkIfCodeExists(commodityCode: String): Boolean = {
-    val canonicalCode: String =
-      if (commodityCode.length > padLimit) commodityCode.substring(0, padLimit)
-      else commodityCode.trim.padTo[Char, String](padLimit, '0').mkString
+  def find(commodityCode: String): Option[CommodityCode] = {
+    val canonicalCode: String = padTo10Digits(commodityCode)
+
+    commodityCodesFromFile.find(_.code == canonicalCode)
+  }
+
+  private def padTo10Digits(input: String): String = {
+    val trimmed = input.trim
+    if (trimmed.length > padLimit) trimmed.substring(0, padLimit)
+    else trimmed.padTo[Char, String](padLimit, '0').mkString
+  }
+
+  private val commodityCodesFromFile: Seq[CommodityCode] = {
+    val url = getClass.getClassLoader.getResource(appConfig.commodityCodePath)
+    val lines = Source.fromURL(url, "UTF-8").getLines()
```

Comment:
Should we be closing this file after reading? This will leave a handle to file open till the JVM exists.

Reply:
From : https://alvinalexander.com/scala/how-to-open-read-text-files-in-scala-cookbook-examples

`Read the file into a string
Use this approach to get all of the lines from the file as one String:

val fileContents = Source.fromFile(filename).getLines.mkString
This approach has the side effect of leaving the file open as long as the JVM is running, but for short-lived shell scripts, this shouldn’t be an issue; the file is closed when the JVM shuts down.

Properly closing the file
To properly close the file, get a reference to the BufferedSource when opening the file, and manually close it when you’re finished with the file:

val bufferedSource = Source.fromFile("example.txt")
for (line <- bufferedSource.getLines) {
    println(line.toUpperCase)
}

bufferedSource.close
For automated methods of closing the file, see the “Loan Pattern” examples in the Discussion.

The getLines method of the Source class returns a scala.collection.Iterator. The iterator returns each line without any newline characters. An iterator has many methods for working with a collection, and for the purposes of working with a file, it works well with the for loop, as shown.

 `

Reply:
I suspect the stream is closed when the iterator finishes. If I try and close it manually I get IOExceptions as the stream is closed

Reply:
This works (not saying it's the best ever code) - 

val src = Source.fromURL(url, "UTF-8")
    val lines = src.getLines()

    val byHeader: Map[String, Int] = split(lines.next()).zipWithIndex.toMap

    try {
      lines
        .map(split)
        .filter(columns => columns(byHeader("leaf")) == "1")
        .map { columns =>
          val commodityCode = columns(byHeader("goods_nomenclature_item_id"))
          val expiry = columns(byHeader("validity_end_date"))
          if (expiry.nonEmpty) {
            val date = LocalDateTime.parse(expiry, dateTimeFormatter).atZone(appConfig.clock.getZone).toInstant
            CommodityCode(commodityCode, Some(date))
          } else {
            CommodityCode(commodityCode)
          }
        }.toSeq
    } finally {
      println("OK I will close the file then")
      src.close()
    }

Without that I can see using `sudo lsof -u aaroncarroll | grep 'commodityCodes'` that it's left open otherwise